### ISSUE TITLE
Hanging signs cause ClassCastException and do not drop with text

### ISSUE THREAD

**Issue by logitechno** [joshuaprince/SilkSigns#1]
When splitting a hanging sign, this error is displayed. The text is not saved
Core: purpur
Version: 1.21
https://mclo.gs/L3C2iHJ

**Comment by joshuaprince** [comment]
Confirmed on latest builds of Spigot, Paper, and Purpur.

This is caused by a bug in Spigot which was added with the item meta changes on 2024-07-06. I filed [SPIGOT-7846](https://hub.spigotmc.org/jira/browse/SPIGOT-7846) since it needs to be fixed there before the error here can be fixed. I don't see any easy workaround for SilkSigns until then. Until it's fixed in Spigot, Paper versions 89 and below (Purpur versions 2267 and below) should be unaffected.
